src/request/plan.rs

```diff
@@ -349,3 +402,60 @@ mod test {
             .for_each(|r| assert!(r.is_err()));
     }
 }
+
+// contains a response and the corresponding keys
+// currently only used for matching keys and values in pessimistic lock requests
+#[derive(Debug, Clone)]
+pub struct ResponseAndKeys<Resp>(Resp, Vec<Key>);
```

Comment:
Could this just be a tuple?

Reply:
I doubt tuples can be ambiguous. It's not intuitive to tell what `(Response, Vec<Key>)` means, IMO.

Reply:
IMO the reading of the tuple is exactly `Response` and `Key`s. To be a bit abstract, I believe that a tuple struct is equivalent to a newtype and similarly should be used if and only if the newtype introduces an invariant over the component types.

Reply:
Makes sense. 
I think one thing that introducing the new type benefits is that the traits are implemented only for this type but not for general `(Resnponse, Vec<Key>)`. So it won't interfere with other uses of `(Resnponse, Vec<Key>)` if we use it for other purposes, or even where we may need different implementations of these traits (though not very likely).

Reply:
Are there other uses of (Response, Vec<Key>) where this impl does not make sense?

Reply:
I can't think of one...